src/controllers/HealthCheckController.ts

```diff
@@ -0,0 +1,69 @@
+import { Connection } from 'mongoose';
+
+export interface IResult {
+  database: {
+    isConnected: boolean;
+    status: number;
+    message: string;
+  };
+  service: {
+    isReady: boolean;
+    status: number;
+    message: string;
+  };
+}
+
+export default class HealthCheckController {
+  constructor(private connection: Connection) {}
+
+  public getServiceStatus(): IResult {
+    return {
+      database: {
+        isConnected: this.isConnected(),
+        status: this.connection.readyState,
+        message: this.databaseStatusMessage(),
+      },
+      service: {
+        isReady: this.isConnected(),
+        status: this.isConnected() ? 1 : 0,
```

Comment:
Would be better if we return true/false in this case? Maybe you can use BANG operator to do this

Reply:
isReady is boolean. status is numeric version of isReady for feature better description current service status,